### PR TITLE
Add tests for empty or malformed recipient gpg key

### LINKED ISSUE #6889: Need Sequoia/redwood test for PGP encryption if no recipients are specified
In Python we have:

```python
        # When trying to encrypt some data without providing any recipient
        # It fails and the right exception is raised
        with pytest.raises(GpgEncryptError) as exc:
            encryption_mgr._encrypt(
                using_keys_with_fingerprints=[],
                plaintext_in="test",
                ciphertext_path_out=tmp_path / "encrypt_fails",
            )
        assert "no terminal at all requested" in str(exc)
```

We should have a similar test (in Rust) that verifies the behavior if no recipients are specified to `redwood::encrypt()`.

### PR DESCRIPTION
**Pull request by rocodes**
## Status

Ready for review 

## Description of Changes

Fixes #6889.

Changes proposed in this pull request:

Add tests for cases of missing or malformed journalist key

Visual review/CI passes

## Deployment 

n/a

### If you made changes to the server application code:

- [x] Linting (`make lint`) and tests (`make test`) pass in the development container (`make rust-lint`, `make rust-test`)

### If you made non-trivial code changes:

- [x] I have written a test plan and validated it for this PR

Choose one of the following:

- [ ] I have opened a PR in the [docs repo](https://github.com/freedomofpress/securedrop-docs) for these changes, or will do so later
- [ ] I would appreciate help with the documentation
- [x] These changes do not require documentation

### CODE REVIEW

**Review by legoktm — COMMENTED** [review]
Awesome! Tests and errors look good. Since most of the code is identical across the tree tests, do you want to consolidate it into one test function that iterates over the 3 cases? There's no fancy parameterization in Rust like pytest has so it'd just be iterating over a vec.

File: redwood/src/lib.rs [hunk 1]
```diff
@@ -244,4 +245,23 @@ mod tests {
         // Verify message is what we put in originally
         assert_eq!("Rust is great 🦀", &plaintext);
     }
+
+    #[test]
+    fn test_encryption_missing_malformed_recipient_key() {
+        let map: HashMap<Vec<String>, &str> = HashMap::from([
+            (vec![], "OpenPGP error: Invalid operation: Neither recipients, passwords, nor session key given"), // missing key
```
Comment (rocodes) [inline redwood/src/lib.rs]: cargo fmt did not have anything to say about this very long line...

File: redwood/src/lib.rs [hunk 2]
```diff
@@ -244,4 +244,34 @@ mod tests {
...
+                vec!["".to_string()],
+                "OpenPGP error: Malformed Cert: No data",
+            ),
+            (
+                vec!["DEADBEEF0123456".to_string()],
+                "OpenPGP error: unexpected EOF",
+            ),
+        ];
+        let tmp: NamedTempFile = NamedTempFile::new().unwrap();
+        for bad_key in bad_keys {
```
Comment (legoktm) [inline redwood/src/lib.rs]: One small nitpick that is hopefully useful for learning purposes: you can unpack the tuple while iterating, so this could be:
```rust
for (recipients, expected_err) in bad keys {
    ...
}
```

instead of having to refer to them as `.0` or `.1`.

**Review by legoktm — DISMISSED** [review]
One tiny note if you want to change it, otherwise LGTM!

File: redwood/src/lib.rs (hunk 2 above)
Comment (rocodes) [reply redwood/src/lib.rs]: tis indeed an improvement, thank you for all the back and forth on this, it has been super helpful :)

Comment (rocodes) [reply redwood/src/lib.rs]: (I'm going to fix it and push cause I'm a weirdo perfectionist)

**Review by legoktm — APPROVED** [review]
Awesome 🚀
